noir-projects/noir-contracts/contracts/key_registry_contract/src/main.nr

```diff
@@ -35,71 +39,59 @@ contract KeyRegistry {
   #[aztec(public)]
   fn rotate_nullifier_public_key(
     address: AztecAddress,
-    new_nullifier_public_key: Field,
+    new_nullifier_public_key: GrumpkinPoint,
   ) {
     assert(
-      new_nullifier_public_key != 0,
+      !new_nullifier_public_key.is_zero(),
       "New nullifier public key must be non-zero"
     );
 
     if (!address.eq(context.msg_sender())) {
       assert_current_call_valid_authwit_public(&mut context, address);
```

Comment:
Nonce? 👀 
Without the nonce, it would not be possible to rotate back to a key that you have used in the past if you are using the authwit. It is a real edge case, so think it is fair enough to ignore, but would add a small comment "Beware that no-nonce in authwit, so cannot send approve the same action again"